### PR TITLE
feat: [ST-3448] Add `hreflang_x_default_lang` setting

### PR DESCRIPTION
**Pull request by bryanzerrudo**
### Purpose/goal of PR (Issue #)

https://wovnio.atlassian.net/browse/ST-3448

See `README` changes for setting overview.

### CODE REVIEW

**Comment by Soichiro725** [comment]
I want to see the purpose of this PR in this description not in the link. (Jira ticket of link doesn't have description. Parent ticket has description. But ...)

**Review by Soichiro725 — APPROVED** [review]

File: src/wovnio/html/HtmlConverter.php [hunk 1]
```diff
@@ -280,6 +280,16 @@ private function insertHreflangTags($html)
             array_push($hreflangTags, '<link rel="alternate" hreflang="' . Lang::iso6391Normalization($lang_code) . '" href="' . $href . '">');
         }
 
+        if (isset($this->store->settings['hreflang_x_default_lang'])) {
+            $x_default_hreflang_regex = "/<link [^>]*hreflang=[\"']?(x-default)[\"']?(\s[^>]*)?\>/iU";
```
Comment (Soichiro725) [inline src/wovnio/html/HtmlConverter.php:284]: I prefer the following regex. I think it works as same way.
```
<link[^>]*hreflang=[\"']?x-default[\"']?[^>]*>
```
- No need to do capture
- `[^>]` can include space
- `*` can be nothing, so no need to add `?`

Comment (bryanzerrudo) [reply src/wovnio/html/HtmlConverter.php]: Fixed in https://github.com/WOVNio/WOVN.php/pull/207/commits/40443223bc9f64743e846c214cbcd596f871688b.

File: README.en.md [hunk 2]
```diff
@@ -932,6 +933,31 @@ outbound_proxy_host = site.com
 outbound_proxy_port = 8080
 ```
 
+#### `hreflang_x_default_lang`
+Configures the language code used to generate link tag with `hreflang="x-default"` attribute.
+
+The language code used must be in `supported_langs`. If the language code is invalid, link tag will not be inserted.
+
+If a link tag with `hreflang="x-default"` attribute exists, this setting does nothing.
```
Comment (Soichiro725) [inline README.en.md:941]: When customer uses `custom_domain` pattern, there is a case we need to overwrite link tag with `hreflang="x-default"` when customer has config to change their default lang URL.

Comment (Soichiro725) [reply README.en.md:941]: For now, I agree with current spec.

**Review by Soichiro725 — APPROVED** [review]

File: test/unit/html/HtmlConverterTest.php [hunk 3]
```diff
@@ -196,6 +196,100 @@ public function testInsertSnippetAndLangTagsWithInsertHreflangsFalse()
         }
     }
 
+    public function testInsertSnippetAndLangTagsWithHreflangXDefaultLang()
+    {
+        $html_cases = array(
+            array (
+                'has setting and html has none - is generated',
```
Comment (bstewart00) [inline test/unit/html/HtmlConverterTest.php:203]:
```suggestion
                'has x-default setting and html has none - is generated',
```

File: test/unit/html/HtmlConverterTest.php [hunk 4]
```diff
@@ -196,6 +196,100 @@ public function testInsertSnippetAndLangTagsWithInsertHreflangsFalse()
...
+                '</head>' .
+                '<body>' .
+                '<a>hello</a>' .
+                '</body>' .
+                '</html>',
+
+                'en'
+            ),
+            array (
+                'has setting and html has existing - is not modified',
```
Comment (bstewart00) [inline test/unit/html/HtmlConverterTest.php:223]:
```suggestion
                'has x-default setting and html has existing - is not modified',
```

File: test/unit/html/HtmlConverterTest.php [hunk 5]
```diff
@@ -196,6 +196,100 @@ public function testInsertSnippetAndLangTagsWithInsertHreflangsFalse()
...
+                '</head>' .
+                '<body>' .
+                '<a>hello</a>' .
+                '</body>' .
+                '</html>',
+
+                'en'
+            ),
+            array (
+                'no setting and html has existing - is not modified',
```
Comment (bstewart00) [inline test/unit/html/HtmlConverterTest.php:250]:
```suggestion
                'no x-default setting and html has existing - is not modified',
```

**Review by bstewart00 — APPROVED** [review]

**Pull request merged**